piccolo_api/crud/endpoints.py

```diff
@@ -820,19 +828,31 @@ async def post_single(
             return Response("Unable to save the resource.", status_code=500)
 
     @apply_validators
-    async def delete_all(
+    async def delete_bulk(
         self, request: Request, params: t.Optional[t.Dict[str, t.Any]] = None
     ) -> Response:
         """
-        Deletes all rows - query parameters are used for filtering.
+        Bulk deletes rows - query parameters are used for filtering.
         """
         params = self._clean_data(params) if params else {}
         split_params = self._split_params(params)
+        split_params_ids = split_params.ids.split(",")
 
         try:
-            query = self._apply_filters(
-                self.table.delete(force=True), split_params
-            )
+            query: t.Union[
+                Select, Count, Objects, Delete
+            ] = self.table.delete()
+            try:
+                ids = [
+                    int(item) if len(item) < len(str(uuid.uuid4())) else item
+                    for item in split_params_ids
+                ]
```

Comment:
I would do this slightly differently - rather than working out whether it's a UUID based on the length, you can do it instead like this:

```python
isinstance(self.table._meta.primary_key, UUID) # where UUID is Piccolo's UUID column type
```

Reply:
This will not work because `split_params_ids` is a list of strings. I can add type annotation to `split_params id` like this `split_params_ids: t.List[str] = split_params.ids.split(",")`. I added `len(str(uuid.uuid4()))` because Postgres `BigInt` is same as `sys.maxsize` which is length 19.

Reply:
What about this:

```python
value_type = self.table._meta.primary_key.value_type
ids = [value_type(item) for item in split_params_ids]
```

Reply:
This works perfectly. Thanks.